Objects/perf_trampoline.c

```diff
@@ -0,0 +1,219 @@
+#include "Python.h"
+#include "pycore_ceval.h"
+#include "pycore_frame.h"
+#include "pycore_interp.h"
+
+#ifdef HAVE_PERF_TRAMPOLINE
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+typedef PyObject *(*py_evaluator)(PyThreadState *, _PyInterpreterFrame *,
+                                  int throwflag);
+typedef PyObject *(*py_trampoline)(py_evaluator, PyThreadState *,
+                                   _PyInterpreterFrame *, int throwflag);
+extern void *_Py_trampoline_func_start;
+extern void *_Py_trampoline_func_end;
+
+struct code_arena_st {
```

Comment:
Document the structure and members in comments. It smells like a singly linked list, that oddly names its traditional `next` pointer `prev`.